etc/dovetail-edge-deploy.yml

```diff
@@ -0,0 +1,865 @@
+# etc/dovetail-edge-deploy.yml
+# Deploys Dovetail Lambda@Edge functions. The primary purpose of the pipeline
+# this creates is to deploy the CloudFormation template that manages the
+# CloudFront distribution for Dovetail, which is itself responsible for
+# managing the Lambda@Edge functions that the distribution uses.
+#
+# Neither this CloudFormation template nor the resulting CodePipelne pipeline
+# handle updating the Lambda function code or versions. Those are managed via
+# the main Infrastructure stack and CD pipeline. This simply deploys those
+# functions to the Edge.
+AWSTemplateFormatVersion: "2010-09-09"
+Description: >
+  Builds a release pipeline via CodePipeline to deploy Dovetail Lambda@Edge
+  functions.
+Metadata:
+  AWS::CloudFormation::Interface:
+    ParameterGroups:
+      - Label:
+          default: GitHub API
+        Parameters:
+          - GitHubToken
+          - GitHubWebhookSecret
+      - Label:
+          default: CloudFormation Stacks
+        Parameters:
+          - InfrastructureNotificationsStackName
+          - StagingCloudFormationStackName
+          - ProductionCloudFormationStackName
+      - Label:
+          default: Staging Dovetail CDN
+        Parameters:
+          - StagingOriginBucket
+          - StagingOriginPath
+          - StagingDistributionDomain
+      - Label:
+          default: Production Dovetail CDN
+        Parameters:
+          - ProductionOriginBucket
+          - ProductionOriginPath
+          - ProductionDistributionDomain
+      - Label:
+          default: Common CDN
+        Parameters:
+          - CertificateArn
+          - BytesLambdaFunctionArn
+          - StitchLambdaFunctionArn
+    ParameterLabels:
+      GitHubToken:
+        default: Access token
+      GitHubWebhookSecret:
+        default: Webhook secret
+      InfrastructureNotificationsStackName:
+        default: Notifications stack name
+      StagingCloudFormationStackName:
+        default: Staging stack name
+      ProductionCloudFormationStackName:
+        default: Production stack name
+      StagingOriginBucket:
+        default: Origin bucket
+      StagingOriginPath:
+        default: Origin path
+      StagingDistributionDomain:
+        default: Distribution domain
+      ProductionOriginBucket:
+        default: Origin bucket
+      ProductionOriginPath:
+        default: Origin path
+      ProductionDistributionDomain:
+        default: Distribution domain
+      BytesLambdaFunctionArn:
+        default: Unqualified Bytes Lambda function ARN
+      StitchLambdaFunctionArn:
+        default: Unqualified Stitch Lambda function ARN
+Parameters:
+  StagingOriginBucket:
+    Default: prx-dovetail.s3.amazonaws.com
```

Comment:
Rather than passing them in here as parameter overrides, we could make a template config JSON file that lives somewhere for the more permanent values. This seems like a good place to start though without having to shuffle files around in S3, and we can switch it up later if this feels brittle.

Reply:
+1